src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/LaunchProfiles/LaunchProfileProjectProperties.cs

```diff
@@ -240,6 +249,32 @@ public async Task SetPropertyValueAsync(string propertyName, string unevaluatedP
                 return;
             }
 
+            // Finally, store it in ILaunchProfile.OtherSettings.
+
+            object? valueObject = null;
+            if (_rule?.GetProperty(propertyName) is BaseProperty property)
+            {
+                valueObject = property switch
+                {
+                    BoolProperty => bool.TryParse(unevaluatedPropertyValue, out bool result) ? result : null,
+                    IntProperty => int.TryParse(unevaluatedPropertyValue, out int result) ? result : null,
+                    StringProperty => unevaluatedPropertyValue,
+                    EnumProperty => unevaluatedPropertyValue,
+                    DynamicEnumProperty => unevaluatedPropertyValue,
+                    _ => throw new InvalidOperationException($"{nameof(LaunchProfileProjectProperties)} does not know how to convert strings to `{property.GetType()}`.")
+                };
+            }
+            else
+            {
+                valueObject = unevaluatedPropertyValue;
+            }
+
+            if (valueObject is not null)
+            {
+                profileUpdateAction = p => p.OtherSettings[propertyName] = valueObject;
```

Comment:
I'm assuming capturing `valueObject` in the closure is fine?

Reply:
Yeah, that should be fine.